pilot/proxy/envoy/config.go

```diff
@@ -784,14 +789,25 @@ func buildInboundListeners(mesh *proxyconfig.MeshConfig, sidecar proxy.Node,
 			}
 
 		default:
-			glog.V(4).Infof("Unsupported inbound protocol %v for port %#v", protocol, servicePort)
+			glog.V(4).Infof("Unsupported inbound protocol %v for port %#v", protocol, listenerPort)
 		}
 
 		if listener != nil {
-			mayApplyInboundAuth(listener, mesh, endpoint.ServicePort.AuthenticationPolicy)
+			mayApplyInboundAuth(listener, mesh, authPolicy)
 			listeners = append(listeners, listener)
 		}
 	}
+	for _, instance := range instances {
+		endpoint := instance.Endpoint
+		servicePort := endpoint.ServicePort
+		protocol := servicePort.Protocol
+		cluster := buildInboundCluster(endpoint.Port, protocol, mesh.ConnectTimeout)
+		clusters = append(clusters, cluster)
+		buildListerner(cluster, instance.Service.Hostname, endpoint.Port, servicePort.AuthenticationPolicy, protocol, endpoint)
+		if servicePort.AuthMigrationPort != nil {
```

Comment:
its twice the work. can you use if-else and call build Listener only once?

Reply:
The first buildListener is for the original port, which should always be created. The second call, already inside a condition, to add an extra listener for the migration port.